client/components/search/index.jsx

```diff
@@ -255,13 +262,13 @@ module.exports = React.createClass( {
 						? this._keyListener.bind( this, 'openSearch' )
 						: null
 					}
-					aria-controls={ 'search-component-' + this.id }
+					aria-controls={ 'search-component-' + this.state.instanceId }
```

Comment:
This is pretty annoying to test, since I get root replaced errors when testing logged out in #3333 — which I think are obscuring any reconciliation errors.

I remember it causing problems with reconciliation, although my memory is hazy. From what I remember, the `id` would be different on the server vs the client, as the counter gets reset when the client loads, but increments forever on the server. I'm hoping that `statics` may help with this, although it's difficult to test.

It could be worth whipping up a `/design` SSR attempt now, just to see what happens with `Search` reconc.